src/DiceRoll.js

```diff
@@ -113,7 +113,8 @@ class DiceRoll {
    *   rolls: ..., // RollResults object or array of roll results
    * });
    *
-   * @param {string|{notation: string, rolls: ResultGroup|Array}} notation The notation to roll
+   * @param {string|{notation: string, rolls: ResultGroup|Array<ResultGroup>}} notation The
```

Comment:
The `rolls` property actually accepts `ResultGroup|Array.<ResultGroup|RollResults|string|number>`.
I set it to just `ResultGroup|Array` because otherwise ESlint didn't like the line length. Splitting it at a length that ESLint did like, meant splitting it between the closing `}`, and the property name, `notation`.
Unfortunately, the JSDoc parser doesn't like it if the property name is on another line.

I'm assuming that Typescript will throw an error if `rolls` is an array that contains any item types that are not specified in the `@param` definition?
If so, that would mean that we do need to include all the possible types like:
```javascript
@param {string|{notation: string, rolls: ResultGroup|Array.<ResultGroup|RollResults|string|number>}} notation The notation to roll
```
Not sure how to handle the splitting of the line though. Maybe if we get ESLint to ignore line lengths for that line.

Reply:
I believe that you do need to use one of the specified types, yeah. I'll likely be working on my project over the weekend, I'll double check that and let you know. If there's any other TypeScript things you want tested lemme know and I'll take a gander.